backend-lib/src/main/rust/lib.rs

```diff
@@ -831,7 +828,7 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_isValidSa
     let res = catch_unwind(&mut env, |env| {
         let _span = tracing::info_span!("RustBackend.isValidSaplingAddress").entered();
         let network = parse_network(network_id as u32)?;
-        let addr = utils::java_string_to_rust(env, &addr);
+        let addr = utils::java_string_to_rust(env, &addr)?;
 
         match Address::decode(&network, &addr) {
             Some(addr) => match addr {
```

Comment:
Consider this helper:
```rust
fn parse_address(env: &JNIEnv, addr: JString<'local>, network_id: jint) -> anyhow::Result<ZcashAddress> {
    let network = parse_network(network_id as u32)?;
    let addr = utils::java_string_to_rust(env, &addr)?;
    Address::decode(&network, &addr).ok_or(anyhow!("Address is for the wrong network"))
}
```
Then lines 830-839 become just:
```rust
            match parse_address(network_id as u32, &addr)? {
                Address::Sapling(_) => Ok(JNI_TRUE),
                Address::Transparent(_) | Address::Unified(_) | Address::Tex(_) => Ok(JNI_FALSE),
            }
```
This can also be used in `isValid{Transparent,Unified,Tex}Address`, saving 15 non-blank lines overall I think.